go/teams/teams.go

```diff
@@ -465,29 +476,33 @@ func (t *Team) sigChangeItem(ctx context.Context, section SCTeamSection, linkTyp
 		return libkb.SigMultiItem{}, err
 	}
 	sig, err := ChangeSig(me, latestLinkID1, t.NextSeqno(), deviceSigningKey, section, linkType, merkleRoot)
-	if err != nil {
-		return libkb.SigMultiItem{}, err
-	}
 
-	signingKey, err := t.keyManager.SigningKey()
-	if err != nil {
-		return libkb.SigMultiItem{}, err
-	}
-	encryptionKey, err := t.keyManager.EncryptionKey()
 	if err != nil {
 		return libkb.SigMultiItem{}, err
 	}
 
-	if section.PerTeamKey != nil {
-		// need a reverse sig
-
-		// set a nil value (not empty) for reverse_sig (fails without this)
-		sig.SetValueAtPath("body.team.per_team_key.reverse_sig", jsonw.NewNil())
-		reverseSig, _, _, err := libkb.SignJSON(sig, signingKey)
+	var signingKey libkb.NaclSigningKeyPair
+	var encryptionKey libkb.NaclDHKeyPair
+	if linkType != libkb.LinkTypeLeave {
```

Comment:
I think you can simplify this.

```
if section.PerTeamKey != nil {
    // get signing key
   // get encrypt key
    // do the sigs, etc.
}
```

In other words, I don't think you need to switch of `linkType` at all here.  What do you think?